### DISCUSSION TITLE
problem sampling from depth texture in OpenGL but not Metal

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3934] [Q&A]
I have some code what writes to a `RenderTarget` with textures attached to the color and depth attachments, and then in an onscreen render, a material samples from the textures.  That is, if I sample from the color texture, all is well.  If I sample from the depth texture, then it works as expected with a Metal back end, but fails with an OpenGL back end, both on the same Mac.  In the OpenGL case, it behaves as if the depth texture is all at value 1.   I'm not sure whether nothing is getting written to the depth texture, or there is a failure to sample from it.  Any suggestions on how to debug that?  In theory, I should be able to look at OpenGL textures using Apple's "OpenGL Profiler" developer tool, but it hasn't been updated for a long time, and lately it does nothing but crash for me.

**Comment by romainguy** [comment]
OpenGL Profiler has stopped being useful years ago unfortunately. Do you have a repro test we could look at?

@pixelflinger @prideout

**Reply by jwwalker** [reply]
It might take me another day to trim it down to a minimum.

**Reply by jwwalker** [reply]
I've attached an Xcode project that attempts to render a little to an offscreen render target, and then sample the depth texture on screen.  It works as expected for Metal, but just shows solid red on OpenGL.

[DepthReader.zip](https://github.com/google/filament/files/6462792/DepthReader.zip)

**Reply by jwwalker** [reply]
I opened issue #3962
